secure_message_v2/application.py

```diff
@@ -50,3 +75,20 @@ def set_default_schema(dbapi_connection, connection_record):
     models.Base.metadata.create_all(engine)
 
     return engine
+
+
+@retry(
```

Comment:
Not convince by the retry need, but as implemented it as required. The requests is hidden in a before_request, this means an endpoint must be called for it to then check UAA is up, I can't see how UAA couldn't be up at that time.

Reply:
If we don't think an explicit retry is required (as requests deal with it appropriately) then happy to not have it retry here. It was to take the opportunity to introduce a good practice pattern, and attempt to build these service to service calls in good way. I agree it's probably a corner case and doesn't add a huge amount of value in this particular scenario.

Ultimately, as long as the calling service (i.e. to this secure-message-v2 service) can handle the negative response gracefully it's probably ok without retrying.

Appreciate the config here is a straw man and was done deliberately, but worth saying that the current `@retry` here isn't graceful, and the compound impact of the requests timeout and the retry timeout makes the request hang longer than is acceptable.
```
secure-message-v2        |   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in secure-message-v2        |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 843, 
secure-message-v2        |     retries = retries.increment(
secure-message-v2        |   File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 519, in secure-message-v2        |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
secure-message-v2        | urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=9080): Max retries exceeded with url: /oauth/token?grant_type=client_credentials&response_type=token&token_format=opaque (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f957d180190>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

Reply:
Again, more work on UAA needed, I provided the framework without optimising

Reply:
As above, happy to pick this up in the stories that request UAA users. Agree that this PR's authentication story does not have a dependency with UAA.